src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java

```diff
@@ -97,19 +97,20 @@ public void saveInBatches(List<Attendance> attendances) {
     }
 
     @Override
-    public void deleteAttendanceByCourseAndDayOfClass(long courseId, Date dateOfClass) {
+    public void deleteAttendanceByCourseAndDayOfClass(long courseId, Date dateOfClass, long sectionId) {
         Validate.notNull(dateOfClass, "The dateOfClass parameter must not be null");
 
         String jpql = "DELETE FROM Attendance WHERE attendanceId IN (" +
                 "select a.attendanceId " +
                 "FROM Attendance a " +
                 "WHERE a.attendanceStudent.canvasCourseId = :courseId " +
                 "AND trunc(a.dateOfClass) = trunc(:dateOfClass)" +
-                ")";
+                "AND a.attendanceStudent.canvasSectionId = :sectionId)";
 
         javax.persistence.Query query = entityManager.createQuery(jpql);
         query.setParameter("courseId", courseId);
         query.setParameter("dateOfClass", dateOfClass, TemporalType.DATE);
+        query.setParameter("sectionId", sectionId);
         int result = query.executeUpdate();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "result" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1481)

Reply:
This issue is minor. I concur with Sonar, and I have asked for additional changes in LK1764, so I'll kick this pull request back for a 30-second change.